app/controllers/atom/listings_controller.rb

```diff
@@ -0,0 +1,37 @@
+class Atom::ListingsController < ApplicationController
+  include ListingsHelper
+
+  before_filter :set_pagination
+
+  respond_to :atom
+  layout false
+
+  # Render atom feed
+  def index
+    @listings = Listing.find_with(params, @current_user, @current_community, @per_page, @page)
+
+    @total_pages = @listings.total_pages
+
+    category = Category.find_by_id(params["category"])
+    @category_label = (category.present? ? "(" + localized_category_label(category) + ")" : "")
+
+    if ["request","offer"].include? params['share_type']
+      listing_type_label = t("listings.index.#{params['share_type']+"s"}")
+    else
+      listing_type_label = t("listings.index.listings")
+    end
+
+    @title = t("listings.index.feed_title",
+               :optional_category => @category_label,
+               :community_name => @current_community.name_with_separator(I18n.locale),
+               :listing_type => listing_type_label)
+    @updated = @listings.first.present? ? @listings.first.updated_at : Time.now
+  end
+
+  private
+
+  def set_pagination
+    @page = params["page"] || 1
```

Comment:
It seems that `@page` and `@per_page` are used only in `index` method, not in the view. So having these as instance variables adds unnecessary complexity.

If you want to keep the `set_pagination` as a separate method, that's ok, but instead of settings the instance variables, you could just return the values.